packages/typegpu-react/src/use-uniform-value.ts

```diff
@@ -1,13 +1,55 @@
 import type * as d from 'typegpu/data';
+import { useRoot } from './root-context.tsx';
+import { useEffect, useMemo, useState } from 'react';
+import type { ValidateUniformSchema } from 'typegpu';
 
 interface UniformValue<TSchema, TValue extends d.Infer<TSchema>> {
   schema: TSchema;
-  value: TValue;
+  value: TValue | undefined;
+  readonly $: d.InferGPU<TSchema>;
 }
 
-export function useUniformValue<TSchema, TValue extends d.Infer<TSchema>>(
-  schema: d.AnyWgslData,
+export function useUniformValue<
+  TSchema extends d.AnyWgslData,
+  TValue extends d.Infer<TSchema>,
+>(
+  schema: ValidateUniformSchema<TSchema>,
   initialValue?: TValue | undefined,
 ): UniformValue<TSchema, TValue> {
-  // TODO: Implement
+  const root = useRoot();
+
+  const [uniformBuffer] = useState(() => {
+    return root.createUniform(
+      schema,
+      initialValue,
+    );
+  });
+
+  useEffect(() => {
+    return () => {
+      uniformBuffer.buffer.destroy();
+    };
+  }, [uniformBuffer]);
+
+  // biome-ignore lint/correctness/useExhaustiveDependencies: This value needs to be stable
+  const uniformValue = useMemo(() => {
+    let currentValue = initialValue;
```

Comment:
Schemas should generally be callable, in which case they return an initial value that matches that schema.

```suggestion
    let currentValue = initialValue ?? schema();
```

Reply:
Thanks for pointing that out! However, with current implementation:

```ts
export function useUniformValue<
  TSchema extends d.AnyWgslData,
  TValue extends d.Infer<TSchema>,
>(
  schema: ValidateUniformSchema<TSchema>,
  initialValue?: TValue | undefined,
): UniformValue<TSchema, TValue> {
```
schema is not callable:

```
This expression is not callable.
  Not all constituents of type 'AnyWgslData | `(Error) ${string}`' are callable.
    Type 'U16' has no call signatures.ts(2349)
```
I'm wondering how can I alter this to simplify the logic.